core/src/main/java/com/sequenceiq/cloudbreak/service/stack/InstanceMetadataService.java

```diff
@@ -49,6 +49,21 @@ public void updateInstanceStatus(Set<InstanceGroup> instanceGroup, com.sequencei
         }
     }
 
+    public Stack saveInstanceAndGetUpdatedStack(Stack stack, List<CloudInstance> cloudInstances) {
+        for (CloudInstance cloudInstance : cloudInstances) {
+            InstanceGroup instanceGroup = getInstanceGroup(stack.getInstanceGroups(), cloudInstance.getTemplate().getGroupName());
+            if (instanceGroup != null) {
+                InstanceMetaData instanceMetaData = new InstanceMetaData();
+                instanceMetaData.setPrivateId(cloudInstance.getTemplate().getPrivateId());
+                instanceMetaData.setInstanceStatus(com.sequenceiq.cloudbreak.api.model.InstanceStatus.REQUESTED);
+                instanceMetaData.setInstanceGroup(instanceGroup);
+                instanceMetaDataRepository.save(instanceMetaData);
+                instanceGroup.getInstanceMetaDataSet().add(instanceMetaData);
+            }
+        }
+        return stack;
```

Comment:
Shouldn't the stack or instancegroup of stack be saved here?

Reply:
I think it shouldn't. The instances are added to instancegroup, so the actual instancegroups are updated (so stack object are updated too). Next stack query will contain the new instances too, because they are saved to the db.